de4dot.code/deobfuscators/Dotfuscator/CflowDecrypter.cs

```diff
@@ -46,24 +46,47 @@ class CflowDecrypter {
 			if (method.Body.Variables.Count == 0)
 				return;
 			var ins = method.Body.Instructions;
-			for (int i = 3; i < ins.Count; i++) {
-				if (ins[i].OpCode.Code != Code.Ldind_I4)
-					continue;
-				if (!ins[i - 1].IsLdloc() && ins[i - 1].OpCode.Code != Code.Ldloca && ins[i - 1].OpCode.Code != Code.Ldloca_S)
+			var Idxs = GetFixIndexs(ins);
+			if (Idxs[0].Count > 0) {
+				foreach (var idx in Idxs[0]) {
+					method.Body.Instructions[idx].OpCode = OpCodes.Nop;
+					method.Body.Instructions[idx].Operand = null;
+				}
+			}
+			if (Idxs[1].Count > 0) {
+				foreach (var idx in Idxs[1]) {
+					method.Body.Instructions[idx].OpCode = OpCodes.Ldloc;
+				}
+			}
+		}
+		public List<int>[] GetFixIndexs(IList<Instruction> instructions) {
```

Comment:
it should return two out variables and return void, also remove 'using System.Collections.Generic;' above

Reply:
ok,thanks .
and i am sorry for not follow the code rules.
please close this ,i will create a new pull request.